rdkPlugins/Storage/source/DynamicMountDetails.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+* If not stated otherwise in this file or this component's LICENSE file the
+* following copyright and licenses apply:
+*
+* Copyright 2022 Sky UK
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "DynamicMountDetails.h"
+#include "StorageHelper.h"
+#include "DobbyRdkPluginUtils.h"
+
+#include <fstream>
+#include <stdio.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/mount.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <sstream>
+#include <fstream>
+#include <dirent.h>
+#include <pwd.h>
+#include <grp.h>
+
+constexpr char MOUNT_TYPE[] = "bind";
+
+// Dynamic mount details constructor
+DynamicMountDetails::DynamicMountDetails(const std::string& rootfsPath,
+                                         const DynamicMountProperties& mountProperties,
+                                         const std::shared_ptr<DobbyRdkPluginUtils> &utils)
+    : mRootfsPath(rootfsPath),
+      mMountProperties(mountProperties),
+      mUtils(utils)
+{
+    AI_LOG_FN_ENTRY();
+
+    AI_LOG_FN_EXIT();
+}
+
+DynamicMountDetails::~DynamicMountDetails()
+{
+    AI_LOG_FN_ENTRY();
+
+    AI_LOG_FN_EXIT();
+}
+
+// -----------------------------------------------------------------------------
+/**
+ *  @brief Checks whether source exists and creates bind mount if found
```

Comment:
In contradiction to comment it doesn't create mount.

Reply:
It creates mount if source exists on the host, e.g. if found.
Reworded for clarity.